File: Semester3/MAP/Seminare/Sem10/Sem10/Factory/TaskContainerFactory.cs
namespace Sem10;

public class TaskContainerFactory : Factory
{
    private static TaskContainerFactory instance;
    
    private TaskContainerFactory() { }

    public static TaskContainerFactory getInstance()
    {
        if (instance == null)
            instance = new TaskContainerFactory();
        return instance;
    }
    
    public AbstractContainer createContainer(Strategy strategy)
    {
        if (Strategy.FIFO == strategy)
            return new Queue();
        return new Stack();
    }
}

File: Semester3/MAP/laboratoare/lab2/TaskManager/src/Factory/TaskContainerFactory.java
package Factory;

import Domain.Container;
import Domain.QuequeContainer;
import Domain.StackContainer;
import Domain.Task;
import Factory.Strategy;

public class TaskContainerFactory implements Factory{
    private static TaskContainerFactory instance = null;

    private TaskContainerFactory() {}

    @Override
    public Container createContainer(Strategy strategy) {
        if(strategy == Strategy.LIFO) {
            return new StackContainer();
        }
        else if(strategy == Strategy.FIFO) {
            return new QuequeContainer();
        }
        return null;
    }

    public static TaskContainerFactory getInstance() {
        if(instance == null){
            instance = new TaskContainerFactory();
        }
        return instance;
    }
}
